src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/controllers/web/DocumentController.kt

```diff
@@ -0,0 +1,46 @@
+package org.vaccineimpact.orderlyweb.controllers.web
+
+import org.vaccineimpact.orderlyweb.ActionContext
+import org.vaccineimpact.orderlyweb.FileSystem
+import org.vaccineimpact.orderlyweb.Files
+import org.vaccineimpact.orderlyweb.controllers.Controller
+import org.vaccineimpact.orderlyweb.db.AppConfig
+import org.vaccineimpact.orderlyweb.db.Config
+import org.vaccineimpact.orderlyweb.errors.MissingParameterError
+import org.vaccineimpact.orderlyweb.errors.OrderlyFileNotFoundError
+import org.vaccineimpact.orderlyweb.guessFileType
+
+class DocumentController(context: ActionContext,
+                         private val config: Config,
+                         private val files: FileSystem) : Controller(context)
+{
+    constructor(context: ActionContext) : this(context, AppConfig(), Files())
+    fun getDocument(): Boolean
+    {
+        val path = context.splat()?.joinToString("/")
+                ?: throw MissingParameterError("Path to document not provided")
+
+        val fileName = context.splat()!!.last()
+
+        val inline = context.queryParams("inline")?.toBoolean() ?: false
+
+        val absoluteFilePath = "${this.config["documents.root"]}/$path"
+
+        if (!files.fileExists(absoluteFilePath))
```

Comment:
Not looking up this file in the db, but just checking whether it exists on disk.

If/when we want to do specific scoped document permissions this will have to change as we'll have to convert filenames and paths to database ids to verify a permission like `document.read/document:1`. 

At the moment we verify scoped report permissions by specifying them to the security filter like `report:<name>/reports.read` where the route contains `:name` as a route param. If we want to integrate something similar with splats we would have to switch over the schema from using numerical primary keys to using the file path as the primary key, so that it's clear from the ul which document is being referred to. 

I don't know whether it's worth changing the schema over now to use full paths for file primary keys, as is already the case for directories?

Reply:
In the modified schema primary keys are paths, not ints, so this comment is obsolete